core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -89,27 +89,79 @@ default void forEach(Consumer<? super T> action) {
     }
 
     /**
-     * Get the only row in the result set.
-     * @throws IllegalStateException if zero or multiple rows are returned
-     * @return the object mapped from the singular row in the results
+     * Returns the only row in the result set. Returns {@code null} if the row itself is
+     * {@code null}.
+     * @throws IllegalStateException if the result set contains zero or multiple rows
+     * @return the only row in the result set.
      */
-    default T findOnly() {
+    default T one() {
         try (ResultIterator<T> iter = iterator()) {
             if (!iter.hasNext()) {
-                throw new IllegalStateException("No element found in 'only'");
+                throw new IllegalStateException("Expected one element, but found none");
             }
 
             final T r = iter.next();
 
             if (iter.hasNext()) {
-                throw new IllegalStateException("Multiple elements found in 'only'");
+                throw new IllegalStateException("Expected one element, but found multiple");
             }
 
             return r;
         }
     }
 
     /**
+     * Returns the only row in the result set, if any. Returns {@code Optional.empty()} if zero
+     * rows are returned, or if the row itself is {@code null}.
+     * @throws IllegalStateException if the result set contains multiple rows
+     * @return the only row in the result set, if any.
+     */
+    default Optional<T> findOne() {
+        try (ResultIterator<T> iter = iterator()) {
+            if (!iter.hasNext()) {
+                return Optional.empty();
+            }
+
+            final T r = iter.next();
+
+            if (iter.hasNext()) {
+                throw new IllegalStateException("Expected zero to one elements, but found multiple");
```

Comment:
would it be useful to include values in err msg like this:
```suggestion
                throw new IllegalStateException(format(
                    "Expected zero to one elements, but found multiple. First two are: %s, %s",
                    r,
                    iter.next()));
```

?

Reply:
It would be useful, but I think it would also risk inadvertent disclosure.  For example if you are reading a user row that contains a password, you would not want that to accidentally get emitted in an error message.

Reply:
I agree with Steven. It would be useful, but the security concern would discourage users with PCI / PHI uses from using Jdbi for fear of exposing sensitive data in logs.